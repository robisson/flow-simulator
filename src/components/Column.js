import React from "react";
import Task from "./Task";

export default function Column({ name, tasks }) {
  return (
    <div className="list">
      <header>{name}</header>
      <ul>
        {tasks && tasks.map(task => <Task key={task.id} name={task.name} />)}
      </ul>
      <footer>{tasks && tasks.length} atividades</footer>
    </div>
  );
}
