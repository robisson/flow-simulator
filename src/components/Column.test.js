import React from "react";
import { mount } from "enzyme";
import Column from "./Column";
import "../../setupTests";

describe("<Column /> component", () => {
  it("should render component correctly with name property and tasks", () => {
    let tasks = [{ id: 1, name: "task" }, { id: 2, name: "task" }];

    let wrapper = mount(<Column name="columnTest" tasks={tasks} />);

    expect(wrapper.props().name).toEqual("columnTest");
    expect(wrapper.props().tasks).toEqual(tasks);

    expect(wrapper).toMatchSnapshot();
  });
});
