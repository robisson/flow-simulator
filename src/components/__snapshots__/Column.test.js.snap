// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Column /> component should render component correctly with name property and tasks 1`] = `ReactWrapper {}`;
